ouroboros-consensus/ouroboros-consensus.cabal

```diff
@@ -326,13 +329,7 @@ library
 
 
   if os(windows)
-    if impl(ghc >= 8.10.2) && impl(ghc < 8.11)
-      -- Fix the Win32 version to be whatever is shipped with GHC.  This is important
-      -- because we have other code that uses the GHC API which requires this specific
-      -- version of the Win32 or constraint solving will fail.
-      build-depends:   Win32 == 2.6.1.0
-    else
-      build-depends:   Win32
```

Comment:
We no longer have these constraints because the hope is we will transition to `ghc-8.10.7` soon.

Reply:
This fixes #3466 :tada:

Reply:
> We no longer have these constraints because the hope is we will transition to ghc-8.10.7 soon.

Can you elaborate here (or better yet: in the PR description) about why this restriction was here in the first place and why it's safe to remove it now. In particular, since we were building with 8.10.4 and now we're building with 8.10.4 and 8.10.7, why is it safe to remove this now?

The existing comment mentions constraint solving and I see a cabal bump here -- did cabal just get better? Etc.

My concern is that we remove this restriction and comment and then someone experiences a constraint solving failure again and doesn't know why.

Reply:
Done!

Reply:
> because we have other code that uses the GHC API which requires this specific version

@nfrisby could it be because your Win32 specific stuff would fail because it's importing `Win32` unqualified and introduces something that becomes available in a later version?